paywall/src/paywall-script/index.ts

```diff
@@ -82,6 +87,27 @@ export class Paywall {
     }
   }
 
+  handleUserInfoEvent = async (info: UserInfo) => {
+    const { readOnlyProvider } = __ENVIRONMENT_VARIABLES__
+    this.userAccountAddress = info.address
+
+    const lockAddresses = Object.keys(this.paywallConfig.locks)
+    const timeStamps = await Promise.all(
+      lockAddresses.map(lockAddress => {
+        return keyExpirationTimestampFor(
+          readOnlyProvider,
+          lockAddress,
+          this.userAccountAddress!
+        )
+      })
+    )
+
+    if (timeStamps.some(val => val > new Date().getTime() / 1000)) {
+      // TODO: communicate to checkout iframe that there is already a valid key
```

Comment:
I think it's better if the checkout iframe does this determination by itself, no?

Reply:
That actually would make it easier, and cleaner due to separation of concerns. That's how we'll handle it then!